tiny_cnn/core/nnp_backend.h

```diff
@@ -88,7 +93,7 @@ class nnp_backend : public math_backend {
           reinterpret_cast<const float*>(&in_data[2]->at(0));
 
       float* output_pointer =
-          reinterpret_cast<float*>(&out_data[0]->at(0));
+          reinterpret_cast<float*>(&out_data[1]->at(0));
 
```

Comment:
@edgarriba
Hmm..Let's check values of weight vector in your debugger if you still get zero vector after fixing dx/dy.